### ISSUE TITLE
Cursor is shown on touch devices

### ISSUE THREAD

**Issue by Mr-Bajs** [ihhub/fheroes2#6824]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Android

### Describe the bug

The mouse cursor is shown on android touch devices. Gameplay wise it doesn't matter but gives a bit of unpolished look and experience.

The cursor should be hidden by default on Android. Most android devices is mobile touch devices but, maybe there should have an option to have android to be shown becouse, there are desktop keyboard/mouse use for some android devices.

Mouse cursor should have an additional option (Hidden) in the settings apart from the current 2 options Black & White and Color. And this setting be default for Android.

### Save file

.

**Comment by oleg-derevenetz** [comment]
> The cursor should be hidden by default on Android.

No, it shouldn't. For example, on adventure map it shows the number of days in travel to the destination tile.

**Comment by Branikolog** [comment]
Without cursor you also wouldn't be able to check, whether you can stand on a tile or if you would be attacked by creature.

**Comment by oleg-derevenetz** [comment]
> Without cursor you also wouldn't be able to check, whether you can stand on a tile or if you would be attacked by creature.

Sure. The mouse cursor performs an informational function in many cases and shouldn't be removed.

**Comment by Mr-Bajs** [comment]
Okej, but I don't think we mean the same thing.

What i'm thinking of the is spear (color) when that show when clicking on fog of war or the menu. It doesn't add anything to the game does on an touch device does it? 

For a to be shown if you would be attacked by standing on a tile, isn't shown until you click or point at at tile. Same as the pathfinding. It's not the that i'm thinking of. If the two images of the mouse cursor the colorsed spear and the black regulor cursur, just those two images where hidden, that would not in any way affect either tileinformation nor the pathfinding. 

I hope i made myself understood.

**Comment by Mr-Bajs** [comment]
> Without cursor you also wouldn't be able to check, whether you can stand on a tile or if you would be attacked by creature.

Sure you would, you just point att the tile with your finger and then you see, no need for a mouse cursor for that?

**Comment by oleg-derevenetz** [comment]
> Sure you would, you just point att the tile with your finger and then you see

And then you see the mouse cursor. It could be a plain horse (no action on that tile), a rearing horse (some action on that tile expected), sword (your hero will be attacked), a number (this tile is not accessible on the current day, but you need to travel that number of days to reach it), or it may be just an arrow (tile is not accessible in principle).

I'm against some "special" logic regarding mouse cursor on touch devices.

**Comment by Mr-Bajs** [comment]
Yes a

> > Sure you would, you just point att the tile with your finger and then you see
> 
> And then you see the mouse cursor. It could be a plain horse (no action on that tile), a rearing horse (some action on that tile expected), sword (your hero will be attacked), a number (this tile is not accessible on the current day, but you need to travel that number of days to reach it), or it may be just an arrow (tile is not accessible in principle).
> 
> I'm against some "special" logic regarding mouse cursor on touch devices.

No not really you don't se the cursor when you click on a tile, when having a hero active, it is hidden whenever there is any useful information like pathfinding or hostile tile. It is that how it works now. 

I don't think there has to be another logic, wouldn't the be an easy option to either have just the 2 mouse cursor images (as shown in the mouse cursor settings, only those 2) to be invisible as a third option?

**Comment by oleg-derevenetz** [comment]
> No not really you don't se the cursor when you click on a tile, when having a hero active, it is hidden whenever there is any useful information like pathfinding or hostile tile. It is that how it works now.

This is not how it works now:

![Screenshot_20230312-205445](https://user-images.githubusercontent.com/32623900/224563240-5e381769-80b5-4788-8c79-e5cc1e4afef6.png)

![Screenshot_20230312-205458](https://user-images.githubusercontent.com/32623900/224563275-eb68c92d-e2d7-42bd-8286-fc6169b7dc2e.png)

Both screenshots are made on my Android device and on both there is a mouse cursor with useful information. On the first screenshot it shows that there is an action tile (rearing horse), on the second screenshot there is a sword (there will be battle on that tile). This information is useful and I don't see a reason to not show it.

**Comment by Mr-Bajs** [comment]
> > No not really you don't se the cursor when you click on a tile, when having a hero active, it is hidden whenever there is any useful information like pathfinding or hostile tile. It is that how it works now.
> 
> This is not how it works now:
> 
> ![Screenshot_20230312-205445](https://user-images.githubusercontent.com/32623900/224563240-5e381769-80b5-4788-8c79-e5cc1e4afef6.png)
> 
> ![Screenshot_20230312-205458](https://user-images.githubusercontent.com/32623900/224563275-eb68c92d-e2d7-42bd-8286-fc6169b7dc2e.png)
> 
> Both screenshots are made on my Android device and on both there is a mouse cursor with useful information. On the first screenshot it shows that there is an action tile (rearing horse), on the second screenshot there is a sword (there will be battle on that tile). This information is useful and I don't see a reason to not show it.

Sure i agree, but that's not what i talk about. I'm talking about just the two default mouse cursors the Black & White and the Color that is shown in the Mouse Cursor settings. Just the black image of a mouse and the colorised spear cursor, just those TWO. 
Not the horse, not the sword, not the pathfinding, not the castle etc etc. 

If you click on the fog of war, then i think you understand what i mean?

**Comment by oleg-derevenetz** [comment]
As I already said, personally I am against any special logic regarding mouse cursor on touch devices, because even on touch devices there may be (and on some game consoles there are) additional controls to control the mouse cursor (i.e. joystick). And this is not a bug in any sense, everything works just as intended. This may be considered as an "improvement", but again - I don't see there any improvement at all, it's just another matter-of-taste nitpick.

**Comment by Mr-Bajs** [comment]
![Screenshot_20230312_192521_fheroes2](https://user-images.githubusercontent.com/93934125/224565011-e9a38965-eb86-4dcf-b3f5-2bba67cb2d99.jpg)

Okey sure. Either case, it is just static mouse image in the shown screenshot just that one. Just so you know what i talk about. That i don't the the use case for that one an android touch device I don't see the problem with an option that having just that image hidden (or no image) as an third option and if you would be using an android with joystick then you would just use either Black & White option or Color. 
 
Anyway, that just my opinion anyway.

**Comment by ihhub** [comment]
I'm converting this issue into a discussion.
